scripts/build.sh

```diff
@@ -11,6 +11,93 @@
 IFS=$'\n\t'
 set -exo pipefail
 
+HELP_TEXT="$(cat <<EOF
+ Compile and build docker images.
+   compile : compile clojurescript
+   -d | --docker : (default: no action) provide either "build" or "push" to
+           build or push a docker image.  "push" will also perform build.
+   -g | --gcr-registry: If this flag is set, will push to the specified GCR repository.
+   -k | --service-account-key-file: (optional) path to a service account key json
+           file. If set, the script will call "gcloud auth activate-service-account".
+           Otherwise, the script will not authenticate with gcloud.
+   -h | --help: print help text.
+ Examples:
+   Jenkins build job should run with all options, for example,
+     ./docker/build.sh compile -d push
+   To build the compile, the image, and push it to a gcr repository.
+     ./docker/build.sh compile -d build -r gcr --project "my-awesome-project"
+\t
+EOF
+)"
+
+# Enable strict evaluation semantics
+set -e
```

Comment:
nit. `set` can use `-u` exits shell with an error when reference an undefined variable, like in `jenkins_build.sh`